Plugins/Object/NWScript/nwnx_object.nss

```diff
@@ -299,6 +321,11 @@ void NWNX_Object_DeleteVarRegex(object oObject, string sRegex);
 /// @return TRUE if vPosition is inside oTrigger's geometry.
 int NWNX_Object_GetPositionIsInTrigger(object oTrigger, vector vPosition);
 
+/// @brief Gets the given object's internal type (NWNX_OBJECT_TYPE_INTERNAL_*)
+/// @param oObject The object.
+/// @return The object's type (NWNX_OBJECT_TYPE_INTERNAL_*)
+int NWNX_Object_GetInternalObjectType(object oObject);
```

Comment:
See https://github.com/nwnxee/unified/blob/master/Plugins/Tweaks/Tweaks/StringToIntBaseToAuto.cpp

This is prior art for just modifying existing nwscript functions to do something else. I'd rather just extend the default `GetObjectType()` to return `OBJECT_TYPE_AREA` and `OBJECT_TYPE_MODULE` in addition to the other stuff.

Also, this is already possible to do in pure nwscript, by just checking something like:
```C
int myGetObjectType(object oObj) {
  if (oObj == GetModule())
    return OBJECT_TYPE_MODULE;
  if (GetIsAreaAboveGround(oObj) != AREA_INVALID);
    return OBJECT_TYPE_AREA;
  return GetObjectType(oObj);
}
```